### ISSUE TITLE
Pyinstaller; Keras 

### ISSUE THREAD

**Issue by VanessaBoucinha** [pyinstaller/pyinstaller#2436]
Hi! I have a problem with pyinstaller and Keras.

I've done a code in python 3.5 and use neural network from keras. When I run the code on my IDLE it run without problems. After I did a exe file with pyinstaller and when I run it appear an error.

===============================
Problem occurred during compilation with the command line below:
"C:\mingw64\bin\g++.exe" -shared -g -DNPY_NO_DEPRECATED_API=NPY_1_7_API_VERSION
-m64 -DMS_WIN64 -I"C:\Users\VBOUCI~1\AppData\Local\Temp\_MEI32842\numpy\core\inc
lude" -I"C:\Users\VBOUCI~1\AppData\Local\Temp\_MEI32842\include" -I"C:\Users\VBO
UCI~1\AppData\Local\Temp\_MEI32842\theano\gof" -L"C:\Users\VBOUCI~1\AppData\Loca
l\Temp\_MEI32842\libs" -L"C:\Users\VBOUCI~1\AppData\Local\Temp\_MEI32842" -o C:\
Users\vboucinha\AppData\Local\Theano\compiledir_Windows-8.1-6.3.9600-Intel64_Fam
ily_6_Model_61_Stepping_4_GenuineIntel-3.5.2-64\cutils_ext\cutils_ext.pyd C:\Use
rs\vboucinha\AppData\Local\Theano\compiledir_Windows-8.1-6.3.9600-Intel64_Family
_6_Model_61_Stepping_4_GenuineIntel-3.5.2-64\cutils_ext\mod.cpp -lpython35
C:\Users\vboucinha\AppData\Local\Theano\compiledir_Windows-8.1-6.3.9600-Intel64_
Family_6_Model_61_Stepping_4_GenuineIntel-3.5.2-64\cutils_ext\mod.cpp:2:40: fata
l error: numpy/npy_3kcompat.h: No such file or directory
compilation terminated.

Traceback (most recent call last):
  File "site-packages\theano\theano\gof\cutils.py", line 306, in <module>
ImportError: No module named 'cutils_ext.cutils_ext'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "site-packages\theano\theano\gof\cutils.py", line 317, in <module>
ImportError: No module named 'cutils_ext.cutils_ext'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "Keras_TratPERSI_v4.py", line 12, in <module>
  File "<frozen importlib._bootstrap>", line 969, in _find_and_load
  File "<frozen importlib._bootstrap>", line 958, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 664, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 634, in _load_backward_compatible
  File "C:\Users\vboucinha\Anaconda3\lib\site-packages\pyinstaller-3.3.dev0+b00c
050-py3.5.egg\PyInstaller\loader\pyimod03_importers.py", line 389, in load_modul
e
    exec(bytecode, module.__dict__)
  File "site-packages\keras\keras\__init__.py", line 2, in <module>
  File "<frozen importlib._bootstrap>", line 969, in _find_and_load
  File "<frozen importlib._bootstrap>", line 958, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 664, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 634, in _load_backward_compatible
  File "C:\Users\vboucinha\Anaconda3\lib\site-packages\pyinstaller-3.3.dev0+b00c
050-py3.5.egg\PyInstaller\loader\pyimod03_importers.py", line 389, in load_modul
e
    exec(bytecode, module.__dict__)
  File "site-packages\keras\keras\backend\__init__.py", line 64, in <module>
  File "<frozen importlib._bootstrap>", line 969, in _find_and_load
  File "<frozen importlib._bootstrap>", line 958, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 664, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 634, in _load_backward_compatible
  File "C:\Users\vboucinha\Anaconda3\lib\site-packages\pyinstaller-3.3.dev0+b00c
050-py3.5.egg\PyInstaller\loader\pyimod03_importers.py", line 389, in load_modul
e
    exec(bytecode, module.__dict__)
  File "site-packages\keras\keras\backend\theano_backend.py", line 1, in <module
>
  File "<frozen importlib._bootstrap>", line 969, in _find_and_load
  File "<frozen importlib._bootstrap>", line 958, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 664, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 634, in _load_backward_compatible
  File "C:\Users\vboucinha\Anaconda3\lib\site-packages\pyinstaller-3.3.dev0+b00c
050-py3.5.egg\PyInstaller\loader\pyimod03_importers.py", line 389, in load_modul
e
    exec(bytecode, module.__dict__)
  File "site-packages\theano\theano\__init__.py", line 79, in <module>
  File "<frozen importlib._bootstrap>", line 969, in _find_and_load
  File "<frozen importlib._bootstrap>", line 958, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 664, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 634, in _load_backward_compatible
  File "C:\Users\vboucinha\Anaconda3\lib\site-packages\pyinstaller-3.3.dev0+b00c
050-py3.5.egg\PyInstaller\loader\pyimod03_importers.py", line 389, in load_modul
e
    exec(bytecode, module.__dict__)
  File "site-packages\theano\theano\scan_module\__init__.py", line 41, in <modul
e>
  File "<frozen importlib._bootstrap>", line 969, in _find_and_load
  File "<frozen importlib._bootstrap>", line 958, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 664, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 634, in _load_backward_compatible
  File "C:\Users\vboucinha\Anaconda3\lib\site-packages\pyinstaller-3.3.dev0+b00c
050-py3.5.egg\PyInstaller\loader\pyimod03_importers.py", line 389, in load_modul
e
    exec(bytecode, module.__dict__)
  File "site-packages\theano\theano\scan_module\scan_opt.py", line 60, in <modul
e>
  File "<frozen importlib._bootstrap>", line 969, in _find_and_load
  File "<frozen importlib._bootstrap>", line 958, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 664, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 634, in _load_backward_compatible
  File "C:\Users\vboucinha\Anaconda3\lib\site-packages\pyinstaller-3.3.dev0+b00c
050-py3.5.egg\PyInstaller\loader\pyimod03_importers.py", line 389, in load_modul
e
    exec(bytecode, module.__dict__)
  File "site-packages\theano\theano\tensor\__init__.py", line 9, in <module>
  File "<frozen importlib._bootstrap>", line 969, in _find_and_load
  File "<frozen importlib._bootstrap>", line 958, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 664, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 634, in _load_backward_compatible
  File "C:\Users\vboucinha\Anaconda3\lib\site-packages\pyinstaller-3.3.dev0+b00c
050-py3.5.egg\PyInstaller\loader\pyimod03_importers.py", line 389, in load_modul
e
    exec(bytecode, module.__dict__)
  File "site-packages\theano\theano\tensor\subtensor.py", line 26, in <module>
  File "<frozen importlib._bootstrap>", line 969, in _find_and_load
  File "<frozen importlib._bootstrap>", line 958, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 664, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 634, in _load_backward_compatible
  File "C:\Users\vboucinha\Anaconda3\lib\site-packages\pyinstaller-3.3.dev0+b00c
050-py3.5.egg\PyInstaller\loader\pyimod03_importers.py", line 389, in load_modul
e
    exec(bytecode, module.__dict__)
  File "site-packages\theano\theano\gof\cutils.py", line 320, in <module>
  File "site-packages\theano\theano\gof\cutils.py", line 285, in compile_cutils
  File "site-packages\theano\theano\gof\cmodule.py", line 2309, in compile_str
Exception: Compilation failed (return status=1): C:\Users\vboucinha\AppData\Loca
l\Theano\compiledir_Windows-8.1-6.3.9600-Intel64_Family_6_Model_61_Stepping_4_Ge
nuineIntel-3.5.2-64\cutils_ext\mod.cpp:2:40: fatal error: numpy/npy_3kcompat.h:
. compilation terminated.
Failed to execute script Keras_TratPERSI_v4

Python 3.5
Pyinstaller 3.3

**Comment by tallforasmurf** [comment]
What a mess. Can anybody understand what might be happening? From the bottom up, what is "script Keras_TratPERSI_v4"?

`pyimod03_importers.py` is a boot loader utility, fine, but why is it (apparently) in a runaway recursion?

And especially, why is there a g++ compile going on during the boot loader startup? PyInstaller never calls g++, has no relation to g++ that I know of.

**Comment by xoviat** [comment]
I think I understand what's going on here. Theano is a python library that dynamically generates, compiles, and runs code because the first two steps significantly reduce the run time of the third step, which is computationally intensive. 

This is not something that PyInstaller supports and I am not sure it is something that can or even should be supported. Calling a compiler from a frozen Python app brings in a whole set of issues that I'm not sure it's reasonable to deal with.

**Comment by VanessaBoucinha** [comment]
Keras_TratPERSI_v4 is the name of the executable file. 

I've done this in linux and I can run the executable file from the script that has keras neural network. Now I need to make it to use in Windows. I have a script that use keras neural network and it works weel in anaconda. I've made a executable file with pyinstaller and I can't run it because of the error above. 

I've replaced the keras neural network sklearn.neural_network.MLPClassifier. The cript in anaconda is work without problems but the executable file gives the same error that in keras.

Can't I use pyinstaller with any library with neural network?
